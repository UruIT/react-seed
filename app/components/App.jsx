import React from 'react';
import MyButton from './MyButton';
import styles from './App.scss';
import { getJson } from '../utils/fetch';

export default class App extends React.Component {
	constructor(props) {
		super(props);
		this.state = {
			joke: ''
		};
	}
	componentDidMount() {
		getJson('https://api.chucknorris.io/jokes/random')
			.then(res => this.setState({
				joke: res.value
			}));
	}
	render() {
		return (
			<div>
				<button className={styles.button} onClick={this.test}>Test</button>
				<MyButton>{ this.state.joke }</MyButton>
			</div>
		);
	}

	test = () => {
		console.error('test')
	}
}
