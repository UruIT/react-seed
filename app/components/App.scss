@import '../styles/box-shadow.scss';

.shadow {
	@include box-shadow(1px, 1px, 4px,#4c4a4a);
}

.container {
	width: 420px;
	padding: 10px;
	margin: 32px auto;
	text-align: center;
}

.button {
	border: 2px solid #000;
	background-color: #fbfb09;
	cursor: pointer;
	font-weight: 600;
	font-size: 2rem;
	width: 200px;
	margin: 2rem;
}
