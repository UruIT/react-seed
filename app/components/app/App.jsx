import React from 'react';
import MyButton from '../button/Button';
import { getJson } from '../../utils/fetch';
import Counter from '../counter/Counter';
import styles from './App.scss';

export default class App extends React.Component {
	constructor(props) {
		super(props);
		this.state = {
			joke: ''
		};
	}
	handleClick = () => {
		return getJson('https://api.chucknorris.io/jokes/random')
			.then(res => this.setState({
				joke: res.value
			}));
	}
	render() {
		let { container, shadow, button } = styles;
		let { joke } = this.state;

		return (
			<div className={`${container} ${shadow}`}>
				<Counter />
				<button className={`${button} ${shadow}`} onClick={this.handleClick}>Test</button>
				{ !!joke && <MyButton text={joke} /> }
			</div>
		);
	}

	test = () => {
		console.error('test')
	}
}
