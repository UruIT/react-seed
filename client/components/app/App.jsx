import React from 'react';
import PropTypes from 'prop-types';
import Button from '../button/Button';
import Counter from '../counter';
import styles from './App.scss';
import { FormattedMessage } from 'react-intl';
import { connect } from 'react-redux';
import { jokeFetchRequested } from './app.action';

const { container, shadow, button } = styles;

export function App({ joke, loading, error, fetchJoke }) {
	let text = (loading && 'loading...') || error || joke;

	return (
		<div className={`${container} ${shadow}`}>
			<Counter />
			<button className={`${button} ${shadow}`} onClick={fetchJoke}>
				<FormattedMessage id="test" defaultMessage="Test" />
			</button>
			<div>
				{ !!text && <Button text={text} /> }
			</div>
		</div>
	);
}

App.propTypes = {
	fetchJoke: PropTypes.func.isRequired,
	joke: PropTypes.string.isRequired,
	loading: PropTypes.bool.isRequired,
	error: PropTypes.string.isRequired
};

const mapStateToProps = ({ app }) => app;
const mapDispatchToProps = dispatch => ({
	fetchJoke: () => dispatch(jokeFetchRequested())
});

export default connect(mapStateToProps, mapDispatchToProps)(App);
