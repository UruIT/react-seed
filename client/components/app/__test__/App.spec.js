import React from 'react';
import { App } from '../App';
import { shallow } from 'enzyme';
import ShallowRenderer from 'react-test-renderer/shallow';

const PROPS = {
	fetchJoke: jest.fn(),
	joke: '',
	loading: false,
	error: ''
};

describe('<App/>', () => {
	const renderer = new ShallowRenderer();

	it('render no joke, no loading and no error', () => {
		const tree = renderer.render(<App {...PROPS} />);

		expect(tree).toMatchSnapshot();
	});

	it('render a joke', () => {
		const tree = renderer.render(<App {...PROPS} joke="joke!" />);

		expect(tree).toMatchSnapshot();
	});

	it('render loading...', () => {
		const tree = renderer.render(<App {...PROPS} loading />);

		expect(tree).toMatchSnapshot();
	});

	it('render error', () => {
		const tree = renderer.render(<App {...PROPS} error="something went wrong!" />);

		expect(tree).toMatchSnapshot();
	});

	it('render loading before joke', () => {
		const tree = renderer.render(<App {...PROPS} joke="joke!" loading />);

		expect(tree).toMatchSnapshot();
	});

	it('render error before joke', () => {
		const tree = renderer.render(<App {...PROPS} joke="joke!" error="error!" />);

		expect(tree).toMatchSnapshot();
	});
});

test('click on button should call props.fetchJoke func', () => {
	const app = shallow(<App {...PROPS} />);

	app.find('button').simulate('click');

	expect(PROPS.fetchJoke).toBeCalled();
});
