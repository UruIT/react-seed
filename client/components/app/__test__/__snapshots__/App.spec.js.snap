// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App/> render a joke 1`] = `
<div
  className="container shadow"
>
  <Connect(Counter) />
  <button
    className="button shadow"
    onClick={[Function]}
  >
    Test
  </button>
  <div>
    <Button
      onClick={[Function]}
      text="joke!"
    />
  </div>
</div>
`;

exports[`<App/> render error 1`] = `
<div
  className="container shadow"
>
  <Connect(Counter) />
  <button
    className="button shadow"
    onClick={[Function]}
  >
    Test
  </button>
  <div>
    <Button
      onClick={[Function]}
      text="something went wrong!"
    />
  </div>
</div>
`;

exports[`<App/> render error before joke 1`] = `
<div
  className="container shadow"
>
  <Connect(Counter) />
  <button
    className="button shadow"
    onClick={[Function]}
  >
    Test
  </button>
  <div>
    <Button
      onClick={[Function]}
      text="error!"
    />
  </div>
</div>
`;

exports[`<App/> render loading before joke 1`] = `
<div
  className="container shadow"
>
  <Connect(Counter) />
  <button
    className="button shadow"
    onClick={[Function]}
  >
    Test
  </button>
  <div>
    <Button
      onClick={[Function]}
      text="loading..."
    />
  </div>
</div>
`;

exports[`<App/> render loading... 1`] = `
<div
  className="container shadow"
>
  <Connect(Counter) />
  <button
    className="button shadow"
    onClick={[Function]}
  >
    Test
  </button>
  <div>
    <Button
      onClick={[Function]}
      text="loading..."
    />
  </div>
</div>
`;

exports[`<App/> render no joke, no loading and no error 1`] = `
<div
  className="container shadow"
>
  <Connect(Counter) />
  <button
    className="button shadow"
    onClick={[Function]}
  >
    Test
  </button>
  <div />
</div>
`;
